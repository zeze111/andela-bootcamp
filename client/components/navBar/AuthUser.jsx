import React from 'react';
import PropTypes from 'prop-types';
import { Dropdown, NavItem } from 'react-materialize';

import { LinkFieldGroup, LinkFieldGroup2 } from '../common/TextFieldGroup';

/** Stateless component to render actions for authenticated users
 *
 * @export {function} Details
 *
 * @param {object} props
 *
 * @returns {null} null
 */
const AuthUser = props => (
  <div>
    <ul className="right hide-on-med-and-down nav-style">
      <LinkFieldGroup
        id="add"
        to="/add-recipe"
        href="/add-recipe"
        dataTool="Add A Recipe"
        icon="add"
      />
      <LinkFieldGroup
        id="favorite"
        to="/user/favorites"
        href="/user/favorites"
        dataTool="Favorites"
        icon="star_border"
      />
      <LinkFieldGroup
        id="recipes"
        to="/user/recipes"
        href="/user/recipes"
        dataTool="My Recipes"
        icon="format_list_bulleted"
      />
      <li>
        <Dropdown
          trigger={
            <div className="caps div-pointer2 username">{props.firstName}
              <i className="material-icons right">arrow_drop_down</i>
            </div>}
        >
          <NavItem
            to="/user/profile"
            href="/user/profile"
            id="profile"
          >
            <div className="text-color"> Profile
            </div>
          </NavItem>
          <NavItem
            id="signOut"
            to="/"
            href="/"
            onClick={props.signout}
          >
            <div className="text-color"> Sign Out
            </div>
          </NavItem>
        </Dropdown>

      </li>
    </ul >
    <ul id="nav-mobile" className="side-nav">
      <LinkFieldGroup2
        to="/add-recipe"
        href="/add-recipe"
        item="Add A Recipe"
        icon="add"
      />
      <LinkFieldGroup2
        to="/user/favorites"
        href="/user/favorites"
        item="Favourites"
        icon="star"
      />
      <LinkFieldGroup2
        to="/user/recipes"
        href="/user/recipes"
        item="My Recipes"
        icon="format_list_bulleted"
      />
      <LinkFieldGroup2
        to="/user/profile"
        href="/user/profile"
        item={props.firstName}
        icon="person"
      />
      <LinkFieldGroup2
        to="/"
        href="/"
        item="Sign Out"
        icon="keyboard_tab"
        onClick={props.signout}
      />
    </ul>
  </div >
);

AuthUser.propTypes = {
  firstName: PropTypes.string.isRequired,
  signout: PropTypes.func.isRequired
};

export default AuthUser;
