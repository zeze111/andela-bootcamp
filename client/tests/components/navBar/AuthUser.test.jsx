import React from 'react';

import AuthUser from '../../../components/navBar/AuthUser';
import store from '../../../store';

const props = {
  signout: jest.fn(() => Promise.resolve()),
  firstName: 'some'
};

describe('Test for AuthUser', () => {
  it('should render component', () => {
    const wrapper = shallow(<AuthUser {...props} store={store}/> );
    expect(wrapper).toMatchSnapshot()
  });
});
