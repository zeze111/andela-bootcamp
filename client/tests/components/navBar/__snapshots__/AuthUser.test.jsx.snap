// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for AuthUser should render component 1`] = `
<div>
  <ul
    className="right hide-on-med-and-down nav-style"
  >
    <LinkFieldGroup
      dataTool="Add A Recipe"
      href="/add-recipe"
      icon="add"
      to="/add-recipe"
    />
    <LinkFieldGroup
      dataTool="Favorites"
      href="/user/favorites"
      icon="star_border"
      to="/user/favorites"
    />
    <LinkFieldGroup
      dataTool="My Recipes"
      href="/user/recipes"
      icon="format_list_bulleted"
      to="/user/recipes"
    />
    <li>
      <Dropdown
        trigger={
          <div
            className="caps div-pointer2 username"
          >
            some
            <i
              className="material-icons right"
            >
              arrow_drop_down
            </i>
          </div>
        }
      >
        <NavItem
          href="/user/profile"
          to="/user/profile"
        >
          <div
            className="text-color"
          >
             Profile
          </div>
        </NavItem>
        <NavItem
          href="/"
          onClick={[MockFunction]}
          to="/"
        >
          <div
            className="text-color"
          >
             Sign Out
          </div>
        </NavItem>
      </Dropdown>
    </li>
  </ul>
  <ul
    className="side-nav"
    id="nav-mobile"
  >
    <LinkFieldGroup2
      href="/add-recipe"
      icon="add"
      item="Add A Recipe"
      to="/add-recipe"
    />
    <LinkFieldGroup2
      href="/user/favorites"
      icon="star"
      item="Favourites"
      to="/user/favorites"
    />
    <LinkFieldGroup2
      href="/user/recipes"
      icon="format_list_bulleted"
      item="My Recipes"
      to="/user/recipes"
    />
    <LinkFieldGroup2
      href="/user/profile"
      icon="person"
      item="some"
      to="/user/profile"
    />
    <LinkFieldGroup2
      href="/"
      icon="keyboard_tab"
      item="Sign Out"
      onClick={[MockFunction]}
      to="/"
    />
  </ul>
</div>
`;
